File: NOVEMBER-2025/30-11-2025/1590.dart
class Solution {
  int minSubarray(List<int> nums, int p) {
    int total = nums.reduce((a, b) => a + b);
    int need = total % p;
    if (need == 0) return 0;

    Map<int, int> map = {0: -1};
    int prefix = 0;
    int res = nums.length;

    for (int i = 0; i < nums.length; i++) {
      prefix = (prefix + nums[i]) % p;
      int target = (prefix - need + p) % p;
      if (map.containsKey(target)) res = (res < i - map[target]!) ? res : i - map[target]!;
      map[prefix] = i;
    }

    return (res == nums.length) ? -1 : res;
  }
}

File: NOVEMBER-2025/30-11-2025/1590.swift
class Solution {
    func minSubarray(_ nums: [Int], _ p: Int) -> Int {
        let total = nums.reduce(0, +)
        let need = total % p
        if need == 0 { return 0 }

        var map = [0: -1]
        var prefix = 0
        var res = nums.count

        for (i, x) in nums.enumerated() {
            prefix = (prefix + x) % p
            let target = (prefix - need + p) % p
            if let idx = map[target] {
                res = min(res, i - idx)
            }
            map[prefix] = i
        }

        return res == nums.count ? -1 : res
    }
}